docs/AIR.md

```diff
@@ -0,0 +1,181 @@
+## AIR: values
+### Scalars
+
+- scalars are fully consistent - have the same value on each peer during a script execution
+- could be an argument of any instruction
+- JSON-based (fold could iterate only over array-based value)
+
+### Streams
+
+- streams are CRDT-like (locally-consistent) - have deterministic execution wrt one peer
+- versioned
+- could be used only by call and fold instructions (more instructions for streams to come)
+- could be turned to scalar (canonicalized)
+
+### Canonicalized streams
+
+- contains an array of elements that was in a stream at the moment of canonicalization
+- canonicalized streams are imutable and fully consistent as scalars
+- has the same algebra as a stream for `match`/`mismatch` and `call` argument
+- has the same algebra as a scalar for `new`
+- has mixed behaviour for with other instructions
+
+## AIR: Instructions
+
+### call
+
+```wasm
+(call <peer_id> (<service name> <service function>) [<arguments list>] <output name>)
```

Comment:
I don't know if it is this PR's scope, but it should be clarified that output is optional and can be a scalar or a stream.

Reply:
ditto